go-zen-chu/nuc-playbook
# Create initialize playbook

## Why

- We need to separate this playbook by 2 steps

1. initialize node (like sshd setting, create ansible user ……)
2. setup k8s and deploy ……

- First step need to be run in root user, on the other, second step need to be run in ansible user created by the first step.

## What

- create initialize playbook which is expected to be run with root user

## QA, Evidence


```
$ ansible-playbook -i hosts.yml initialize.yml         (git)-[init-setup] 
SSH password: 
[WARNING]: Invalid characters were found in group names but not replaced, use -vvvv to see details

PLAY [inits] ******************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************
[WARNING]: Platform linux on host node3 is using the discovered Python interpreter at /usr/bin/python, but future installation
of another Python interpreter could change the meaning of that path. See
https://docs.ansible.com/ansible/2.10/reference_appendices/interpreter_discovery.html for more information.
ok: [node3]
[WARNING]: Platform linux on host node2 is using the discovered Python interpreter at /usr/bin/python, but future installation
of another Python interpreter could change the meaning of that path. See
https://docs.ansible.com/ansible/2.10/reference_appendices/interpreter_discovery.html for more information.
ok: [node2]

TASK [setup : update pacman cache (pacman -Sy)] *******************************************************************************
changed: [node2]
changed: [node3]

TASK [setup : install openssh (includes sshd)] ********************************************************************************
ok: [node3]
ok: [node2]

TASK [setup : install sudo] ***************************************************************************************************
ok: [node3]
ok: [node2]

TASK [setup : Make sure we have a 'wheel' group] ******************************************************************************
ok: [node3]
ok: [node2]

TASK [setup : Make ansible user] **********************************************************************************************
[WARNING]: The input password appears not to have been hashed. The 'password' argument must be encrypted for this module to
work properly.
changed: [node3]
changed: [node2]

TASK [setup : Change port of sshd for security] *******************************************************************************
changed: [node3]
changed: [node2]

TASK [setup : Allow only wheel users to ssh] **********************************************************************************
changed: [node3]
changed: [node2]

TASK [setup : Allow wheel group to have passwordless sudo] ********************************************************************
changed: [node3]
changed: [node2]

TASK [setup : Disallow root SSH access] ***************************************************************************************
changed: [node3]
changed: [node2]

TASK [setup : Update timesyncd config (main NTP)] *****************************************************************************
changed: [node3]
changed: [node2]

TASK [setup : Update timesyncd config (fallback NTP)] *************************************************************************
changed: [node3]
changed: [node2]

TASK [setup : Restart systemd-timesyncd] **************************************************************************************
changed: [node3]
changed: [node2]

TASK [setup : install parted] *************************************************************************************************
ok: [node3]
ok: [node2]

TASK [setup : install which] **************************************************************************************************
ok: [node3]
ok: [node2]

TASK [setup : install vim] ****************************************************************************************************
ok: [node3]
ok: [node2]

TASK [setup : install git] ****************************************************************************************************
changed: [node3]
changed: [node2]

TASK [setup : install zsh] ****************************************************************************************************
ok: [node3]
ok: [node2]

TASK [setup : copy bash profile for ansible user] *****************************************************************************
changed: [node3]
changed: [node2]

RUNNING HANDLER [setup : restart sshd] ****************************************************************************************
changed: [node3]
changed: [node2]

PLAY RECAP ********************************************************************************************************************
node2                      : ok=20   changed=12   unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
node3                      : ok=20   changed=12   unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```
